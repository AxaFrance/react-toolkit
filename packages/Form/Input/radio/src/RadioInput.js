import React from 'react';
import PropTypes from 'prop-types';

import {
  Field,
  FieldConstants as Constants,
} from '@axa-fr/react-toolkit-form-core';
import { InputManager } from '@axa-fr/react-toolkit-core';

import Radio from './Radio';
import RadioModes from './RadioModes';

const propTypes = {
  ...Constants.propTypes,
  options: PropTypes.array.isRequired,
  value: PropTypes.string,
  mode: PropTypes.oneOf([
    RadioModes.default,
    RadioModes.classic,
    RadioModes.inline,
  ]),
};

const defaultProps = {
  ...Constants.defaultProps,
  value: '',
  mode: RadioModes.default,
};

const RadioInput = props => {
  const {
    mode,
    messageType,
    message,
    className,
    classModifier,
    id,
    name,
    isVisible,
    options,
    classNameContainerLabel,
    classNameContainerInput,
    label,
    forceDisplayMessage,
    value,
    onChange,
    readOnly,
    disabled,
    children,
    ...otherProps
  } = props;
  if (!isVisible) {
    return null;
  }
  let rowModifier = classModifier;
  if (mode === RadioModes.classic) {
    rowModifier += ' label-top';
  }
  const newOptions = InputManager.getOptionsWithId(options);
  const firstId = InputManager.getFirstId(newOptions);
  return (
    <Field
      label={label}
      name={name}
      id={firstId}
      message={message}
      messageType={messageType}
      isVisible={isVisible}
      forceDisplayMessage={forceDisplayMessage}
      className={className}
      classModifier={rowModifier}
      classNameContainerLabel={classNameContainerLabel}
      classNameContainerInput={classNameContainerInput}>
      <Radio
        name={name}
        id={id}
        options={newOptions}
        value={value}
        mode={mode}
        onChange={onChange}
        readOnly={readOnly}
        disabled={disabled}
        classModifier={classModifier}
        {...otherProps}
      />
      {children}
    </Field>
  );
};

RadioInput.propTypes = propTypes;
RadioInput.defaultProps = defaultProps;

export default RadioInput;
