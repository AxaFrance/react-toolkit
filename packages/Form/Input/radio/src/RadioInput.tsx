import React, { ComponentProps } from 'react';
import { Field } from '@axa-fr/react-toolkit-form-core';
import { InputManager } from '@axa-fr/react-toolkit-core';
import Radio from './Radio';

import RadioModes from './RadioModes';

type RadioInputProps = ComponentProps<typeof Field> &
  ComponentProps<typeof Radio>;

const RadioInput: React.FC<RadioInputProps> = ({
  mode = RadioModes.default,
  messageType,
  message,
  className,
  classModifier,
  isVisible,
  options,
  classNameContainerLabel,
  classNameContainerInput,
  label,
  forceDisplayMessage,
  children,
  ...radioProps
}) => {
  const fieldClassModifier = `${classModifier ?? ''}${
    classModifier && mode === RadioModes.classic ? ' ' : ''
  }${mode === RadioModes.classic ? 'label-top' : ''}`;
  const newOptions = InputManager.getOptionsWithId(options);
  const firstId = InputManager.getFirstId(newOptions);
  const radioClassModifier = `${classModifier ?? ''}${
    classModifier && forceDisplayMessage && messageType ? ' ' : ''
  }${forceDisplayMessage && messageType ? `${messageType}` : ''}`;
  return (
    <Field
      label={label}
      id={firstId}
      message={message}
      messageType={messageType}
      isVisible={isVisible}
      forceDisplayMessage={forceDisplayMessage}
      className={className}
      classModifier={fieldClassModifier}
      classNameContainerLabel={classNameContainerLabel}
      classNameContainerInput={classNameContainerInput}>
      <Radio
        options={newOptions}
        mode={mode}
        classModifier={radioClassModifier}
        className={className}
        {...radioProps}
      />
      {children}
    </Field>
  );
};

export default RadioInput;
