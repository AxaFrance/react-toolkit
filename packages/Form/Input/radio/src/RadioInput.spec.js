import React from 'react';
import renderer from 'react-test-renderer';
import ReactTestUtils, { act } from 'react-dom/test-utils';
import ReactDOM from 'react-dom';
import RadioInput from './RadioInput';

describe('<RadioInput />', () => {
  let container;

  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    document.body.removeChild(container);
    container = null;
  });

  it('renders RadioInput correctly', () => {
    const tree = renderer
      .create(
        <RadioInput
          label="Label"
          id="id"
          name="name"
          options={[
            { label: 'Option 1', value: '1', id: 'option1_id' },
            { label: 'Option 2', value: '2', id: 'option2_id' },
          ]}
          onChange={() => {}}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should send the id, name and value on change', () => {
    const onChange = jest.fn();
    const Element = (
      <RadioInput
        label="Label"
        id="id"
        name="name"
        options={[
          { label: 'Option 1', value: '1', id: 'option1_id' },
          { label: 'Option 2', value: '2', id: 'option2_id' },
        ]}
        onChange={onChange}
      />
    );

    act(() => {
      ReactDOM.render(Element, container);
      ReactTestUtils.Simulate.change(container.querySelector('#option1_id'));
    });

    expect(onChange.mock.calls).toHaveLength(1);
    expect(onChange.mock.calls[0][0]).toEqual({
      id: 'option1_id',
      name: 'name',
      value: '1',
    });
  });
});
