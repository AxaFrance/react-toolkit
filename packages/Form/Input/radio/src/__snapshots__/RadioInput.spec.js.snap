// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioInput /> renders RadioInput correctly 1`] = `
<div
  className="row af-form__group"
>
  <div
    className="col-md-2"
  >
    <label
      className="af-form__group-label"
      htmlFor=""
    >
      Label
    </label>
  </div>
  <div
    className="col-md-10"
  >
    <div
      className="af-form__radio-custom"
    >
      <input
        autoFocus={null}
        checked={false}
        className="af-form__input-radio"
        disabled={false}
        id="option1_id"
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        readOnly={false}
        tabIndex={null}
        type="radio"
        value="1"
      />
      <label
        className="af-form__label"
        htmlFor="option1_id"
      >
        <span
          className="af-form__description"
        >
          Option 1
        </span>
      </label>
    </div>
    <div
      className="af-form__radio-custom"
    >
      <input
        autoFocus={null}
        checked={false}
        className="af-form__input-radio"
        disabled={false}
        id="option2_id"
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        readOnly={false}
        tabIndex={null}
        type="radio"
        value="2"
      />
      <label
        className="af-form__label"
        htmlFor="option2_id"
      >
        <span
          className="af-form__description"
        >
          Option 2
        </span>
      </label>
    </div>
  </div>
</div>
`;
